src/v/archival/ntp_archiver_service.cc

```diff
@@ -356,9 +356,14 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::wait_all_scheduled_uploads(
         if (_partition->archival_meta_stm()) {
             retry_chain_node rc_node(
               _manifest_upload_timeout, _initial_backoff, &parent);
-            if (!co_await _partition->archival_meta_stm()->add_segments(
-                  _manifest, rc_node)) {
-                vlog(ctxlog.warn, "archival metadata STM update failed");
+            auto result
+              = co_await _partition->archival_meta_stm()->add_segments(
+                _manifest, rc_node);
+            if (!result && result != cluster::errc::not_leader) {
```

Comment:
maybe we should log error on 'warn' level here?
IIRC the problem was caused by logging in case when the replica is no longer a leader. This is expected behaviour but I think that we still need some information in the log about this.
All other errors are actual errors and need to be logged on 'error' or 'warn' levels.